search/search_repository.go

```diff
@@ -365,6 +447,108 @@ func (r *GormSearchRepository) SearchFullText(ctx context.Context, rawSearchStri
 	return result, count, nil
 }
 
+func (r *GormSearchRepository) listItemsFromDB(ctx context.Context, criteria criteria.Expression, start *int, limit *int) ([]workitem.WorkItemStorage, uint64, error) {
+	where, parameters, compileError := workitem.Compile(criteria)
+	if compileError != nil {
+		return nil, 0, errors.NewBadParameterError("expression", criteria)
+	}
+
+	db := r.db.Model(&workitem.WorkItemStorage{}).Where(where, parameters...)
+	orgDB := db
+	if start != nil {
+		if *start < 0 {
+			return nil, 0, errors.NewBadParameterError("start", *start)
+		}
+		db = db.Offset(*start)
+	}
+	if limit != nil {
+		if *limit <= 0 {
+			return nil, 0, errors.NewBadParameterError("limit", *limit)
+		}
+		db = db.Limit(*limit)
+	}
+
+	db = db.Select("count(*) over () as cnt2 , *").Order("execution_order desc")
+
+	rows, err := db.Rows()
+	if err != nil {
+		return nil, 0, errs.WithStack(err)
+	}
+	defer rows.Close()
+
+	result := []workitem.WorkItemStorage{}
+	columns, err := rows.Columns()
+	if err != nil {
+		return nil, 0, errors.NewInternalError(ctx, err)
+	}
+
+	// need to set up a result for Scan() in order to extract total count.
+	var count uint64
+	var ignore interface{}
+	columnValues := make([]interface{}, len(columns))
+
+	for index := range columnValues {
+		columnValues[index] = &ignore
+	}
+	columnValues[0] = &count
+	first := true
+
+	for rows.Next() {
+		value := workitem.WorkItemStorage{}
+		db.ScanRows(rows, &value)
+		if first {
+			first = false
+			if err = rows.Scan(columnValues...); err != nil {
+				return nil, 0, errors.NewInternalError(ctx, err)
+			}
+		}
+		result = append(result, value)
+
+	}
+	if first {
```

Comment:
Everytime a search query finds no results will lead to the control entering this block, and firing another query. 
Would it be expensive ?